### DISCUSSION TITLE
Create and Read Database Config

### DISCUSSION THREAD

**Discussion by treyhendon** [rajanadar/VaultSharp#361] [Q&A]
Is it possible to create and read the `database/config` api?

Specifically, I'm trying to recreate Step 2 in this Postgres Database Setup example: https://developer.hashicorp.com/vault/docs/secrets/databases/postgresql#setup

```
vault write database/config/my-postgresql-database \
    plugin_name="postgresql-database-plugin" \
    allowed_roles="my-role" \
    connection_url="postgresql://{{username}}:{{password}}@localhost:5432/database-name" \
    username="vaultuser" \
    password="vaultpass" \
    password_authentication="scram-sha-256"
```

**Comment by rajanadar** [comment]
hi @treyhendon 

Vault is interacted in two ways by its actors.

 - The "Administrative" way whereby auth methods are enabled, secret engines are enables, database plugins are enabled, connection is setup, roles are setup etc. This step is done by privileged sys admins etc.
 - The "Consumption" way whereby client applications then read secrets, generate credentials etc. This is done by less privileges systems.

In general VaultSharp is meant for option 2, since whoever wants to administer vault would typically use vault ui or vault cli, and not write an app utilizing vault sharp.

However, since the uber goal of VaultSharp is to mimic every API of vault, what I do is give preference to the consumption APIs and ensure they are all there. And for the administrative apis, add them when a need arises.

Long story short, that particular API doesn't exist in VaultSharp YET for the above reasons.

Let me know if you would like to have that API, and I'll add it in the upcoming release in a week.

**Comment by rajanadar** [comment]
will be solved in #362
